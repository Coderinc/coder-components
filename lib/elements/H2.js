import PropTypes from 'prop-types';
import styled from 'styled-components';

const H2 = styled.h2`
  color: ${({ theme, color }) => (color ? theme[color] : theme.primaryFont)};
  font-size: 1.5rem;
  line-height: 2.25rem;
  font-weight: ${({ light }) => (light ? 'lighter' : 'bold')};
`;

H2.propTypes = {
  light: PropTypes.bool,
};

H2.defaultProps = {
  light: false,
};

export default H2;
