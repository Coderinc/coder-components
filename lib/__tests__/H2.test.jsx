import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import H2 from '../elements/H2';

describe('H2', () => {
  it('renders correctly', () => {
    const component = renderer.create(<H2>Lorem Ipsum</H2>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('light font', () => {
    const component = renderer.create(<H2 light>Lorem Ipsum</H2>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});