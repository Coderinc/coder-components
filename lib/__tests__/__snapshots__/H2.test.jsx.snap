// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H2 light font 1`] = `
.c0 {
  font-size: 1.5rem;
  line-height: 2.25rem;
  font-weight: lighter;
}

<h2
  className="c0"
>
  Lorem Ipsum
</h2>
`;

exports[`H2 renders correctly 1`] = `
.c0 {
  font-size: 1.5rem;
  line-height: 2.25rem;
  font-weight: bold;
}

<h2
  className="c0"
>
  Lorem Ipsum
</h2>
`;
